ledger-api/grpc-definitions/com/daml/ledger/api/v1/command_service.proto

```diff
@@ -82,21 +92,69 @@ message SubmitAndWaitRequest {
 
 }
 
+// TODO(v2) How do we communicate this information when we actually want to return an error?
+message SubmitAndWaitResponse {
+  // The completion offset for the submission that triggered this response.
+  // This offset can be used as the submission rank for subsequent command submissions.
+  // Optional
+  // TODO(v2) Make it required for responses with status code `OK` once all Daml ledgers support it?
+  string completion_offset = 1;
+
+  // Whether this response is a definite answer for the deduplication and rank guarantees,
+  // as defined in ``commands.proto``.
+  // Always true if the status code is `OK`.
+  // TODO(v2) Should this be inverted to `cancelled` as what is proposed for the read/write service API?
+  bool definite_answer = 2;
+}
+
 message SubmitAndWaitForTransactionIdResponse {
   // The id of the transaction that resulted from the submitted command.
   // Must be a valid LedgerString (as described in ``value.proto``).
   // Required
+  // TODO(v2) Why is this field required? In ``completion.proto``, the `transaction_id` is optional.
   string transaction_id = 1;
+
+  // The completion offset for the submission that triggered this response
+  // This offset can be used as the submission rank for subsequent command submissions.
+  // Optional
+  // TODO(v2) Make it required for responses with status code `OK` once all Daml ledgers support it.
+  string completion_offset = 2;
```

Comment:
In case of a `DUPLICATE` completion for submission `sid2` that was caused by the submission `sid1`. Do we get the completion offset for `sid2` or the one for `sid1` here?

Reply:
This is the completion offset that triggered this response. The details of the other submission are in the error details. I'll separately sketch how I imagine to pass error details back to the application.